src/views/Pools/components/PoolCard/Modals/StakeModal.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { Modal, Text, Flex, Image, Button, Slider, BalanceInput, AutoRenewIcon } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+import { BASE_EXCHANGE_URL } from 'config'
+import { useSousStake } from 'hooks/useStake'
+import { useSousUnstake } from 'hooks/useUnstake'
+import useTheme from 'hooks/useTheme'
+import BigNumber from 'bignumber.js'
+import { getFullDisplayBalance, formatNumber } from 'utils/formatBalance'
+import useToast from 'hooks/useToast'
+import { Pool } from 'state/types'
+
+interface StakeModalProps {
+  isBnbPool: boolean
+  pool: Pool
+  stakingMax: BigNumber
+  stakingTokenPrice: number
+  isRemovingStake?: boolean
+  onDismiss?: () => void
+}
+
+const StyledButton = styled(Button)`
+  flex-grow: 1;
+`
+
+const StakeModal: React.FC<StakeModalProps> = ({
+  isBnbPool,
+  pool,
+  stakingMax,
+  stakingTokenPrice,
+  isRemovingStake = false,
+  onDismiss,
+}) => {
+  const { sousId, stakingToken, earningToken } = pool
+  const TranslateString = useI18n()
+  const { theme } = useTheme()
+
+  const { onStake } = useSousStake(sousId, isBnbPool)
+  const { onUnstake } = useSousUnstake(sousId)
+  const { toastSuccess, toastError } = useToast()
+
+  const [pendingTx, setPendingTx] = useState(false)
+  const [stakeAmount, setStakeAmount] = useState('')
+  const [percent, setPercent] = useState(0)
+
+  const usdValueStaked = stakeAmount && formatNumber(parseFloat(stakeAmount) * stakingTokenPrice)
+
+  const handleStakeInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value ? event.target.value : '0'
```

Comment:
Just commenting about locales and input elements, feel free to ignore. 

ok, I figured out wtf with the input and commas/points.
The way this works is that in my locale apparently comma is the "official separator" (although I disagree, stupid locales). So when event with `.` is fired it for some reason erases `event.target.value` (and `event.currentTarget.value`). This sucks cause I can't even use numpad at all.
The fix (?) is to not do ternary here, just yeet `event.target.value`, it works fine as shown in video (BalanceInput on top is just doing `console.log(event.target.value)`).

And setting `0` here explicitly is not necessary since there is placeholder on input already.

BUT - this would involve extra checks cause with my locale if I put `0.` the `new BigNumber(event.target.value)` would be `NaN`, so it would require a check for `inputBigNumber.isNan()`, which is ugly... 

So I guess let's yeet it like that, would make sense to throughly test it separately and come up with some input handling that works well in different locales. Europeans might get mad but I guess its fine, lets hope I'm the only one in europe who uses point for decimal separator. 😓 

https://user-images.githubusercontent.com/81824236/116302358-bbfa8080-a7a9-11eb-85a4-70305a3daef1.mp4

Reply:
Yea nice one - great to have this intel if we start getting feedback from the community team bringing this behaviour up